src/main/java/dev/sigstore/fulcio/client/FulcioClient.java

```diff
@@ -122,26 +115,7 @@ public CertificateResponse SigningCert(CertificateRequest cr, String bearerToken
     if (sctHeader == null) {
       throw new IOException("no signed certificate timestamps were found in response from Fulcio");
     }
-    byte[] sct = Base64.getDecoder().decode(sctHeader);
 
-    System.out.println(new String(sct));
-
-    CertificateFactory cf = CertificateFactory.getInstance("X.509");
-    ArrayList<X509Certificate> certList = new ArrayList<>();
-    PemReader pemReader = new PemReader(new InputStreamReader(resp.getContent()));
-    while (true) {
-      PemReader.Section section = pemReader.readNextSection();
-      if (section == null) {
-        break;
-      }
-
-      byte[] certBytes = section.getBase64DecodedBytes();
-      certList.add((X509Certificate) cf.generateCertificate(new ByteArrayInputStream(certBytes)));
-    }
-    if (certList.isEmpty()) {
-      throw new IOException("no certificates were found in response from Fulcio");
-    }
-
-    return new CertificateResponse(cf.generateCertPath(certList), sct);
+    return SigningCertificate.newSigningCertificate(resp.getContent(), sctHeader);
```

Comment:
This looks like an unclosed `InputStream`, doesn't it?

Reply:
definitely is, lemme go fix that too.

Reply:
so this method signature now changed to accept string, it's probably okay to store the pem chain in memory for now.